blocks/main_loop.py

```diff
@@ -231,6 +231,12 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                if other not in ['before_batch', 'after_batch']:
+                    raise ValueError("I'm being compared to something \
+                                      I don't know!")
```

Comment:
No, the comparison is made between a Callback instance and a string. So other is just a string, self == other would work too I guess.